### ISSUE TITLE
bug in profile unset property functionality

### ISSUE THREAD

**Issue by odedva** [klaviyo/klaviyo-api-python#35]
Hi there. 
i think that there is a but that causes regression with the unset profile tag functionality. it used to work for a few weeks and then this change caused it to break:
https://github.com/klaviyo/klaviyo-api-python/commit/2ba4b481e8639066b204ef8cf97c5f9a7e8f76a5#r124652203
when trying to use the update_profile method, i get this error:
```
pydantic.error_wrappers.ValidationError: 1 validation error for ProfilePartialUpdateQuery
data -> meta -> patch_properties -> unset
  value is not a valid dict (type=type_error.dict)
```
if i'm putting a dictionary, then i get an API error from your servers...
```
HTTP response body: {"errors":[{"id":"53bad77c-ed7a-497c-8ee1-33161572b398","status":400,"code":"invalid",
"title":"Invalid input.","detail":"Unset must be a string or an array of strings (the property keys to unset).",
"source":{"pointer":"/data/attributes/meta/patch_properties/unset"},"meta":{}}]}
```

i'm doing something like this:
```
                profile_id = 'dummy-profile-id'
                profile_update_data = {
                    'type': 'profile',
                    'id': profile_id,
                    'attributes': {},
                    'meta': {
                        'patch_properties': {
                            'unset': 'check_integration',
                        },
                    },
                }
                self.api.Profiles.update_profile(
                    profile_id,
                    ProfilePartialUpdateQuery(data=profile_update_data),
                )
```

will appreciate your help

**Comment by jon-batscha** [comment]
Hey Oded,


Thanks for flagging this issue, I was able to replicate it on my end.

I'll try to get a fix out asap.


Cheers,
Jon

**Comment by jon-batscha** [comment]
Following up to say I just released a fix in the latest version, 5.1.1

Your code should now work if you upgrade to v5.1.1:


That said, I have a tiny recommendation to improve your code above to be slightly simpler/in-line with our recommended practices:

In your code example:

```python
                profile_id = 'dummy-profile-id'
                profile_update_data = {
                    'type': 'profile',
                    'id': profile_id,
                    'attributes': {},
                    'meta': {
                        'patch_properties': {
                            'unset': 'check_integration',
                        },
                    },
                }
                self.api.Profiles.update_profile(
                    profile_id,
                    ProfilePartialUpdateQuery(data=profile_update_data),
                )
```

You are passing in ProfilePartialUpdateQuery, and setting the data inside the parens. Though this technically works, I recommend to instead construct the fully-formed `body` param (*including* `data`) as a dict, and pass it in as-is.

This is how we suggest doing this in the [function signature](https://github.com/klaviyo/klaviyo-api-python#update-profile) section of the README (pasted in below):

```python
## Positional Arguments

# id | str
# body | dict

klaviyo.Profiles.update_profile(id, body)
```

For the structure of any given param, such as `body`, you can follow the link to the language-agnostic API docs from the function signature link above. In the case of `update_profile`, that would be [this page](https://developers.klaviyo.com/en/reference/update_profile).

For this example, that would mean your new call would look as follows:

```python
profile_id = 'dummy-profile-id'

profile_update_data = {
    'data': {
        'type': 'profile',
        'id': profile_id,
        'attributes': {},
        'meta': {
            'patch_properties': {
                'unset': 'check_integration',
            },
        },
    },
}

self.api.Profiles.update_profile(
    profile_id,
    profile_update_data,
)
```

Thanks again for finding this bug and reaching out, and apologies for any difficulties you came across in integrating with Klaviyo. We're always looking to improve our developer experience, and we can't do it without feedback from our developer community. Definitely keep in touch regarding any future bugs/challenges/suggestion you come across.


Cheers,
Jon
